extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts

```diff
@@ -0,0 +1,174 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { ProxyIdentifier, createProxyIdentifier } from '@theia/plugin-ext/lib/api/rpc-protocol';
+
+export interface CheApiPlugin {
+
+}
+
+export interface CheApiMain {
+    $currentWorkspace(): Promise<Workspace>;
+
+    $getFactoryById(id: string): Promise<Factory>;
+}
+
+export interface Factory {
+    workspace: WorkspaceConfig;
+    ide?: {
+        onAppLoaded?: {
+            actions?: FactoryAction[]
+        };
+        onProjectsLoaded?: {
+            actions?: FactoryAction[]
+        };
+        onAppClosed?: {
+            actions?: FactoryAction[]
+        };
+    }
+}
+
+export interface FactoryAction {
```

Comment:
There is no explaining documentationexplaining. I don't understand what fields mean.